tests/forwarding/watch_port_test.cc

```diff
@@ -675,13 +734,124 @@ TEST_P(WatchPortTestFixture, VerifyBasicWatchPortAction) {
+  ASSERT_OK_AND_ASSIGN(const std::string port_name,
+                       gutil::FindOrStatus(port_name_per_port_id,
+                                           absl::StrCat(selected_port_id)));
+
+  // Set the system in critical state by triggering a fake alarm in P4RT.
+  ASSERT_TRUE(GetParam().set_critical_alarm.has_value());
+  ASSERT_OK((*GetParam().set_critical_alarm)(testbed.Sut()));
+
+  // Set admin down from control switch side (since sut is in critical state and
+  // write operations are disabled) and verify watch port action kicks in.
+  ASSERT_OK(SetInterfaceAdminState(*control_gnmi_stub_, port_name,
+                                   AdminState::kDown));
+
+  // TODO: Adding watch port action causes unexpected traffic
+  // loss. Remove after the bug in OrchAgent is fixed.
+  absl::SleepFor(absl::Seconds(5));
+
+  // Clear the counters before the test.
+  test_data_.ClearReceivedPackets();
+
+  // Send 5000 packets and check for packet distribution.
+  ASSERT_OK(SendNPacketsToSut(kNumTestPackets, test_config, members,
+                              controller_port_ids, GetIrP4Info(),
+                              *control_p4_session_, environment));
+  test_data_.total_packets_sent = kNumTestPackets;
+
+  // Wait for packets from the SUT to arrive.
+  absl::SleepFor(kDurationToWaitForPackets);
+
+  // For the test configuration, check the output distribution.
+  {
+    absl::MutexLock lock(&test_data_.mutex);
+    TestInputOutput& test = test_data_.input_output_per_packet[test_config_key];
+    EXPECT_EQ(test.output.size(), test_data_.total_packets_sent)
+        << "Mismatch in expected: " << test_data_.total_packets_sent
+        << " and actual: " << test.output.size() << "packets received for "
+        << DescribeTestConfig(test_config);
+
+    ASSERT_OK(VerifyGroupMembersFromP4Read(*sut_p4_session_, GetIrP4Info(),
+                                           kGroupId, members));
+
+    // Count the received packets and create the expected_member_ports for admin
+    // down case to verify received packets.
+    ASSERT_OK_AND_ASSIGN(auto num_packets_per_port,
+                         CountNumPacketsPerPort(test.output));
+    absl::flat_hash_set<int> expected_member_ports =
+        CreateExpectedMemberPorts(members);
+    expected_member_ports.erase(selected_port_id);
+    ASSERT_OK(VerifyGroupMembersFromReceiveTraffic(num_packets_per_port,
+                                                   expected_member_ports));
+
+    PrettyPrintDistribution(test_config, test, test_data_, members,
+                            num_packets_per_port);
+  }
+};
 
 // Bring up/down the only ActionProfileGroup member and verify traffic is
 // forwarded/dropped.
 TEST_P(WatchPortTestFixture, VerifyWatchPortActionForSingleMember) {
   thinkit::TestEnvironment& environment =
       GetParam().testbed->GetMirrorTestbed().Environment();
+  environment.SetTestCaseID("60da7a07-1217-4d63-9716-1219d62065ff");
```

Comment:
Pls remove all test case ids in a separate commit for this test and in future tests.